src/app_service/service/contacts/service.nim

```diff
@@ -297,25 +297,47 @@ QtObject:
       else:
         contact.blocked = false
 
-      let response = status_contacts.addContact(contact.id, contact.name)
+      let response = status_contacts.sendContactRequest(contact.id, message)
       if(not response.error.isNil):
         let msg = response.error.message
-        error "error adding contact ", msg
+        error "error sending contact request", msg
         return
       self.saveContact(contact)
       self.events.emit(SIGNAL_CONTACT_ADDED, ContactArgs(contactId: contact.id))
     except Exception as e:
-      error "an error occurred while edding contact ", msg=e.msg
+      error "an error occurred while sending contact request", msg=e.msg
 
-  proc rejectContactRequest*(self: Service, publicKey: string) =
-    var contact = self.getContactById(publicKey)
-    contact.removed = true
+  proc acceptContactRequest*(self: Service, publicKey: string) =
+    try:
+      # NOTE: publicKey used for accepting last request
+      let response = status_contacts.acceptLatestContactRequestForContact(publicKey)
+      if(not response.error.isNil):
+        let msg = response.error.message
+        error "error accepting contact request", msg
+        return
+
+      # TODO: does we need this logic on both (sendContactRequest and acceptContactRequest) sides?
```

Comment:
What about this TODO ?

Reply:
This TODO is for farther contact request implementation but mostly depend on yours first question